sympy/solvers/solvers.py

```diff
@@ -1557,7 +1557,22 @@ def _expand(p):
                         try:
                             t = Dummy('t')
                             iv = _solve(u - t, symbol, **flags)
+                            temp_soln = soln
                             soln = list(ordered({i.subs(t, s) for i in iv for s in soln}))
+                            count_nan = 0
+                            if S.NaN in soln:
+                                count_nan = soln.count(S.NaN)
+                                # try getting a solution for the equation
+                            if count_nan != 0:
+                                # heuristic -01(check whether the root satisfies the equation)
+                                index_list = []
+                                for index, s in enumerate(soln):
+                                    if s == S.NaN:
+                                        index_list.append(index)
+                                for i in index_list:
+                                    if checksol(f, symbol, temp_soln[i]):
```

Comment:
As I've said, no point. This will hardly ever happen - your equation is a very, very special case. Even for the more general case of the equation `x**(1/x)-c`: this strategy will not work for any value of `c` that's not +/-1. You might as well check if any integers in the range [1,10] are solutions - they might be in some cases. But what's the point if in general it's not true and not even likely to happen.